lib/Serialization/Deserialization.cpp

```diff
@@ -1660,6 +1660,18 @@ ModuleFile::resolveCrossReference(ModuleDecl *baseModule, uint32_t pathLen) {
             "cross-reference to generic param for non-generic type",
             pathTrace, getXRefDeclNameForError());
       }
+
+      unsigned currentDepth = paramList->getDepth();
+      if (currentDepth < depth) {
+        return llvm::make_error<XRefError>(
+            "a containing type has been made non-generic",
+            pathTrace, getXRefDeclNameForError());
+      }
+      while (currentDepth > depth) {
+        paramList = paramList->getOuterParameters();
+        --currentDepth;
```

Comment:
Is it worth asserting that the GenericTypeParamDecl you find below has the right depth and index?

Reply:
That seems like a good idea for the AST verifier in general. It's set up by construction that the depth matches the GenericParamList's notion of depth, but the GenericParamList and the GenericTypeParamDecl could be out of sync. I'll do that in a follow-up PR.